### PR TITLE
Inclusion of Router::reverse()

### PR DESCRIPTION
**Pull request by dreamingmind**
There is more editing I would like to do here but it seems best to see if the overall restructuring of this section, and the example I have used are acceptable.

At line 1337 I'm proposing a new glossary term, pass value. But there may be some existing place in the docs to reference for an explanation of this term. There is even this https://book.cakephp.org/4/en/development/routing.html#passed-arguments on the same page.

At line 1378 I'm proposing a new :ref: link which would go to this heading https://book.cakephp.org/4/en/controllers/request-response.html#request-parameters

### CODE REVIEW

File: en/development/routing.rst [hunk 1]
```diff
@@ -1299,11 +1298,46 @@ application renaming URLs. However, if you defined your link like::
...
+
+It is also useful when the destination is unknown but follows a well
+defined pattern::
+
+    $this->Html->link(
+        'View',
+        ['controller' => $controller, 'action' => 'view', $id]
+    );
+
+Elements included un-keyed in the array are treated as
```
Comment (markstory) [inline en/development/routing.rst:1336]:
```suggestion
Elements with numeric keys are treated as
```

File: en/development/routing.rst [hunk 2]
```diff
@@ -1338,6 +1372,78 @@ You can also use any of the special route elements when generating URLs:
 * ``_name`` Name of route. If you have setup named routes, you can use this key
   to specify it.
 
+Using ``Router::reverse()``
+---------------------------
+
+``Router::reverse()`` allows you to use the :ref:`request-parameters' in cases
```
Comment (markstory) [inline en/development/routing.rst:1378]:
```suggestion
``Router::reverse()`` allows you to use the :ref:`request-parameters` in cases
```

File: en/development/routing.rst [hunk 3]
```diff
@@ -1338,6 +1372,78 @@ You can also use any of the special route elements when generating URLs:
...
+where the current url is the basis for the destination, with some modification
```
Comment (markstory) [inline en/development/routing.rst:1379]:
```suggestion
where the current URL is the basis for the destination, with some modification
```

File: en/development/routing.rst [hunk 4]
```diff
@@ -1338,6 +1372,78 @@ You can also use any of the special route elements when generating URLs:
...
+and the elements of the current url are unpredictable.
```
Comment (markstory) [inline en/development/routing.rst:1380]:
```suggestion
and the elements of the current URL are unpredictable.
```

File: en/development/routing.rst [hunk 5]
```diff
@@ -1338,6 +1372,78 @@ You can also use any of the special route elements when generating URLs:
...
+    $params = Hash::insert($params, '?.published', 1);
+    echo $this->Html->link(__('Published'), Router::reverse($params));
+
+    /* Prepare url for All */
+    $params = Hash::remove($params, '?.published');
+    echo $this->Html->link(__('All'), Router::reverse($params));
+
+The links generated by these method calls would include one or two pass
+parameters depending on the structure of the current URL. And the code
+would work for any future url, for example, if you started using
```
Comment (markstory) [inline en/development/routing.rst:1417]:
```suggestion
would work for any future URL, for example, if you started using
```

**Comment by dreamingmind** [comment]
Great! I'll just sort out the final edits I want to propose... back soon

**Comment by dreamingmind** [comment]
These are the last edits I wanted to propose

File: en/development/routing.rst [hunk 6]
```diff
@@ -1293,23 +1293,26 @@ And then later decide that ``/articles`` should really be called
...
 
-    $this->Html->link(
-        'View',
-        Router::reverse(Router::getRequest()->getAttributes('params'))
-    );
+    //'Router::reverse()' operates on the request parameters array
+    //and will produce a url string, valid input for `link()`
+
+    $request_params = Router::getRequest()->getAttributes('params');
+    $this->Html->link('View', Router::reverse($request_params));
```
Comment (markstory) [inline en/development/routing.rst:1309]:
```suggestion
    $requestParams = Router::getRequest()->getAttributes('params');
    $this->Html->link('View', Router::reverse($requestParams));
```

To match code conventions used in the book.

**Comment by dreamingmind** [comment]
That's it for me! Thanks!

**Comment by markstory** [comment]
Thank you 👍

**Pull request merged**